### ISSUE TITLE
Next 12 Error ESM in production, works locally 

### ISSUE THREAD

**Issue by mateusmirandaalmeida** [vercel/next.js#30347]
### What version of Next.js are you using?

12.0.0

### What version of Node.js are you using?

v14.17.5

### What browser are you using?

Chrome

### What operating system are you using?

Linux PopOS

### How are you deploying your application?

Vercel

### Describe the Bug

Apparently a conflict with Swiper ESM package.
In version 11 of next.js it worked.
we import normal:
import {Swiper, SwiperSlide} from 'swiper/react';
after upgrading to version 12 of the next, in development it works, but in vercel it doesn't

### Expected Behavior

info  - Loaded env from /var/task/.env.local
2021-10-26T18:13:28.652Z	673e2ff4-a116-4da1-87f0-59e86b91fbf5	ERROR	Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"SyntaxError: Named export 'useEffect' not found. The requested module 'react' is a CommonJS module, which may not support all module.exports as named exports.\nCommonJS modules can always be imported via the default export, for example using:\n\nimport pkg from 'react';\nconst { useEffect, useLayoutEffect } = pkg;\n","reason":{"errorType":"SyntaxError","errorMessage":"Named export 'useEffect' not found. The requested module 'react' is a CommonJS module, which may not support all module.exports as named exports.\nCommonJS modules can always be imported via the default export, for example using:\n\nimport pkg from 'react';\nconst { useEffect, useLayoutEffect } = pkg;\n","stack":["file:///var/task/node_modules/swiper/react/use-isomorphic-layout-effect.js:1","import { useEffect, useLayoutEffect } from 'react';","         ^^^^^^^^^","SyntaxError: Named export 'useEffect' not found. The requested module 'react' is a CommonJS module, which may not support all module.exports as named exports.","CommonJS modules can always be imported via the default export, for example using:","","import pkg from 'react';","const { useEffect, useLayoutEffect } = pkg;","","    at ModuleJob._instantiate (internal/modules/esm/module_job.js:121:21)","    at async ModuleJob.run (internal/modules/esm/module_job.js:166:5)","    at async Loader.import (internal/modules/esm/loader.js:178:24)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: SyntaxError: Named export 'useEffect' not found. The requested module 'react' is a CommonJS module, which may not support all module.exports as named exports.","CommonJS modules can always be imported via the default export, for example using:","","import pkg from 'react';","const { useEffect, useLayoutEffect } = pkg;","","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:412:35)","    at processPromiseRejections (internal/process/promises.js:245:33)","    at processTicksAndRejections (internal/process/task_queues.js:96:32)"]}
Unknown application error occurred

### To Reproduce

1 - Install swiper@7
2 - Upgrade Next.js to version 12
3 - Publish in Vercel

**Comment by johnrackles** [comment]
> If it only happens on vercel, but works with local `next build && next start` you might try that as workaround too:
> 
> ```js
> // next.config.js
> module.exports = {
>   outputFileTracing: false,
> }
> ```
> 
> but just guessing. I also seems like you all might have different problems...

This seems to have done the trick for my problem, thank you very much!

**Comment by ijjk** [comment]
@johnrackles did you need to add `outputFileTracing: false` in the latest version of Next.js `v12.0.2`?

**Comment by revskill10** [comment]
@ijjk In our case, we need `outputFileTracing` has we need to read certFile for https. Any other workaround, too ?
Here's my log

![image](https://user-images.githubusercontent.com/1390196/140050878-0e5c0b95-09ac-488c-9f00-1ca561a7af03.png)

I'm using Next 12.0.2

**Comment by ijjk** [comment]
@revskill10 that sounds related to https://github.com/vercel/next.js/issues/30750 which is fixed in v12.0.3-canary.2 of Next.js

**Comment by johnrackles** [comment]
> @johnrackles did you need to add `outputFileTracing: false` in the latest version of Next.js `v12.0.2`?

yes, I just tried again and it still happenes. I'll try with v12.0.3-canary.2 and report back

**Comment by timneutkens** [comment]
12.0.3 is out now 👍

**Comment by JoviDeCroock** [comment]
Hey y'all,

I quickly dove into this and used the reproduction so kindly provided by @rayriffy (thank you 🙌 )! So upgrading `@preact/compat` to 17.0.3 removes the tracing issue. However it surfaces another issue, Next seems to have an issue with deduping packages.

This is mainly an assumption because I've seen it before when porting `prefresh` to next 12, I saw that [as soon as I imported Preact from normal preact](https://github.com/preactjs/prefresh/pull/401/files#diff-3f85e0d557c4e4b3a66b70e3722f42b7357d52f2eb6b5bdebc5940bfdaa9f4c7R1) it would have two duplicate imports of our options hooks.
